include/moveit_middleware_benchmark/scenarios/scenario_perception_pipeline.hpp

```diff
@@ -0,0 +1,163 @@
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Cihat Kurtuluş Altıparmak
+   Description: Benchmarking module to compare the effects of middlewares
+   against perception pipeline
+ */
+
+#pragma once
+
+#include <rclcpp/rclcpp.hpp>
+#include <benchmark/benchmark.h>
+#include <moveit/move_group_interface/move_group_interface.h>
+#include <memory>
+
+#include <moveit/moveit_cpp/moveit_cpp.h>
+#include <moveit/moveit_cpp/planning_component.h>
+
+#include <geometry_msgs/msg/pose.hpp>
+#include <nav_msgs/msg/path.hpp>
+#include <dynmsg/msg_parser.hpp>
+#include <dynmsg/typesupport.hpp>
+#include <dynmsg/yaml_utils.hpp>
+
+#include <ament_index_cpp/get_package_share_directory.hpp>
+
+namespace
+{
+const std::string PLANNING_GROUP = "panda_arm";
+const std::string PACKAGE_SHARE_DIRECTORY = ament_index_cpp::get_package_share_directory("moveit_middleware_benchmark");
+const std::string TEST_CASES_YAML_FILE =
+    PACKAGE_SHARE_DIRECTORY + "/config/scenario_perception_pipeline_test_cases.yaml";
+}  // namespace
+
+namespace moveit
+{
+namespace middleware_benchmark
+{
+
+using moveit::planning_interface::MoveGroupInterface;
+
+class ScenarioPerceptionPipeline
+{
+public:
+  /** \brief Constructor for interface to send goal poses
+   *  \param [in] move_group_interface_ptr The move_group_interface for sending goal poses
+   */
+  ScenarioPerceptionPipeline(std::shared_ptr<MoveGroupInterface> move_group_interface_ptr);
+
+  /** \brief Given a \e pose_list, sends poses in pose_list to plan,
+   *  if the pipeline in \e sendTargetPose returns true,
+   *  then success_number is increased.
+   *  If the pipeline in \e sendTargetPose returns false,
+   *  failure_number is increased.
+   *  \param [in] test_case The pose list to benchmark
+   *  \return success_number and failure_number in tuple respectively
+   */
+  std::tuple<int, int> runTestCase(const nav_msgs::msg::Path& pose_list);
+
+  /** \brief Given a \e target_pose, sends this pose in order to plan via move_group_interface,
+   *  if the planning is successful,
+   *  planning message is sent to execute trajectory. Then it is returned true.
+   *  If the planning is failed, the execution is ignored and then it is returned false
+   *  \param [in] target_pose The pose to plan
+   *  \return if the planning successful, returns true, or false
+   */
+  bool sendTargetPose(const geometry_msgs::msg::Pose& target_pose);
+
+private:
+  /** move_group_interface_ptr to communicate with move_grouo_server */
+  std::shared_ptr<MoveGroupInterface> move_group_interface_ptr_;
+};
+
+class ScenarioPerceptionPipelineFixture : public benchmark::Fixture
+{
+protected:
```

Comment:
Nitpick: I'd put the protected class members after the public ones

Reply:
Fixed in https://github.com/CihatAltiparmak/moveit_middleware_benchmark/pull/5/commits/916a595a895dfc8febc655658a490f4051c8efa4 commit